tools/buildsys/src/deps.rs

```diff
@@ -0,0 +1,241 @@
+// TODO - fix all clippy issues
+#![allow(clippy::all)]
+
+use crate::manifest::ManifestInfo;
+use std::collections::{HashMap, HashSet};
+use std::error::Error;
+use std::fmt::{Debug, Display, Formatter};
+use std::path::{Path, PathBuf};
+
+/// TODO - replace with Snafu even though I hate it.
+#[derive(Debug, Default, Clone)]
+pub struct TempError(String);
+
+impl Display for TempError {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        Display::fmt(&self.0, f)
+    }
+}
+
+impl Error for TempError {}
+
+type Result<T> = std::result::Result<T, TempError>;
+
+#[derive(Debug, Clone, Copy, Eq, PartialEq, Ord, PartialOrd)]
+pub enum ManifestType {
+    Package,
+    Kit,
+}
+
+pub struct ManifestIdentity {
+    /// The name of the kit or package as found in `Cargo.toml`.
+    _manifest_name: String,
+    /// Whether this is a kit or a package.
+    _manifest_type: ManifestType,
+    /// The path to the manifest.
+    _manifest_path: PathBuf,
+    // // TODO - keep only the subset of info that is required.
+    // manifest: ManifestInfo,
+}
+
+pub struct Graph {
+    /// The starting point of this graph, i.e., the name of the kit or package that was the starting
```

Comment:
Curious whether you looked at the `guppy` crate.

Based on [this example](https://github.com/guppy-rs/guppy/blob/main/guppy/examples/remove_dev_only.rs), it looks like we could take the `cargo metadata` output (possibly generated once, at the start of the build), build a graph, find our crate in it, then iterate over it with a custom resolver.

Reply:
Done!